src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -176,41 +158,34 @@ private void emitFunctionalInterfaceGetter(String fiName, String javaName) {
             """);
     }
 
-    private void emitFieldGetter(String vhConstant, String javaName, Class<?> type) {
+    private void emitFieldGetter(String javaName, Type varType, long offset) {
         String seg = safeParameterName("seg");
+        Class<?> type = Utils.carrierFor(varType);
+        MemoryLayout layout = Type.layoutFor(varType).get();
         appendIndentedLines(STR."""
             public static \{type.getSimpleName()} \{javaName}$get(MemorySegment \{seg}) {
-                return (\{type.getName()}) \{vhConstant}.get(\{seg}, 0L);
+                return \{seg}.get(\{layoutString(1, layout)}, \{offset});
```

Comment:
what do you mean "using layout string directly" ?

Reply:
I mean, we don't use a reference to a layout constant we generated elsewhere, but always generate the layout string inline

Reply:
still don't get it :-) The `layout` variable is a local variable in this method?

Reply:
Ok, I guess you meant this as an observation: since it's a primitive layout, we just emit JAVA_XYZ inline. This is different from what we do for offset (and size). If you prefer to add a layout constant too, I can add (for readability).

Reply:
Right sorry, it was just an observation/making sure I'm understanding the code correctly.